### DISCUSSION TITLE
Bearer Token based auth in nuxt3

### DISCUSSION THREAD

**Discussion by kirtan403** [nuxt-community/auth-module#1752] [Q&A]
Is there a better way to manually implement bearer token based auth in a nuxt3 project? (until the auth module for nuxt3 officially comes out). The reason is I have a few clients that wants to leverage nuxt3 as the RC is already out and nuxt contentv2 is already out. The only stopping way for me is auth. 

Does anyone has any template or some guidance on which I can start implementing bearer token-based auth in nuxt3? And then replace that with the auth module when it lands.

Anything would be helpful. Thanks?

**Comment by Stiropor** [comment]
I used this as base logic: https://github.com/Atinux/discuss/blob/main/composables/github.ts

Didn't take me long to make it work on Laravel Sanctum with bearer token, login, logout, get user data and added computed isLoggedIn.

EDIT: this is a quick mockup if it helps anyone: https://gist.github.com/Stiropor/94c125e2a77c1a50977d73ddae064a1e

**Reply by kirtan403** [reply]
This really looked pretty good solution. But I guess pinia needs to be used to persist the token?

**Reply by kirtan403** [reply]
I'm not sure about cookies solution...
